src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,387 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.integration
+
+/-!
+
+# Vector valued measures
+
+This file defines vector valued measures, which are σ-additive functions from a set to a add monoid
+`M` such that it maps the empty set and non-measurable sets to zero. In the case
+that `M = ℝ`, we called the vector measure a signed measure and write `signed_measure α`.
+Similarly, when `M = ℂ`, we call the measure a complex measure and write `complex_measure α`.
+
+## Main definitions
+
+* `vector_measure` is a vector valued, σ-additive function that maps the empty
+  and non-measurable set to zero.
+
+## Implementation notes
+
+We require all non-measurable sets to be mapped to zero in order for the extensionality lemma
+to only compare the underlying functions for measurable sets.
+
+We use `has_sum` instead of `tsum` in the definition of vector measures in comparison to `measure`
+since this provides summablity.
+
+## Tags
+
+vector measure, signed measure, complex measure
+-/
+
+noncomputable theory
+
+open_locale classical big_operators nnreal ennreal
+
+namespace measure_theory
+
+variables {α β : Type*} [measurable_space α]
+
+/-- A vector measure on a measurable space `α` is a σ-additive `M`-valued function (for some `M`
+an add monoid) such that the empty set and non-measurable sets are mapped to zero. -/
+structure vector_measure (α : Type*) [measurable_space α]
+  (M : Type*) [add_comm_monoid M] [topological_space M] :=
+(measure_of : set α → M)
+(empty : measure_of ∅ = 0)
+(not_measurable ⦃i : set α⦄ : ¬ measurable_set i → measure_of i = 0)
+(m_Union ⦃f : ℕ → set α⦄ :
+  (∀ i, measurable_set (f i)) → pairwise (disjoint on f) →
+  has_sum (λ i, measure_of (f i)) (measure_of (⋃ i, f i)))
+
+-- A `signed_measure` is a `ℝ`-vector measure.
+notation `signed_measure ` α := vector_measure α ℝ
+-- A `complex_measure` is a `ℂ`-vector_measure.
+notation `complex_measure ` α := vector_measure α ℂ
+
+open set measure_theory
+
+namespace vector_measure
+
+variables {M : Type*} [add_comm_monoid M] [topological_space M]
+
+instance : has_coe_to_fun (vector_measure α M) :=
+⟨λ _, set α → M, vector_measure.measure_of⟩
+
+@[simp]
+lemma apply (v : vector_measure α M) (i : set α) : v.measure_of i = v i := rfl
```

Comment:
```suggestion
@[simp]
lemma measure_of_eq_coe (v : vector_measure α M) (i : set α) : v.measure_of = v := rfl
```

Reply:
I think we should name it `of_eq_coe` in that case since we decided to change `measure_of_foo` to `of_foo` in one of the collapsed discussions above

Reply:
I don't agree - here you're literally saying `vector_measure.measure_of` is the same as `coe`. In the collapses discussion, you're saying "what word should we use to describe `⇑v i`", where `of` is fine.

This is analogous to `monoid_hom.to_fun_eq_coe`

Reply:
See also `measure_theory.outer_measure.measure_of_eq_coe`

Reply:
I see, I will take you word for it then :)